database/sqlite.go

```diff
@@ -87,31 +87,29 @@ func sqliteConnection(logger log.Logger, path string) *sqlite {
 // TestSQLiteDB is a wrapper around sql.DB for SQLite connections designed for tests to provide
 // a clean database for each testcase.  Callers should cleanup with Close() when finished.
 type TestSQLiteDB struct {
-	DB *sql.DB
-
-	dir string // temp dir created for sqlite files
-
+	*sql.DB
+	dir      string // temp dir created for sqlite files
 	shutdown func() // context shutdown func
 }
 
 func (r *TestSQLiteDB) Close() error {
 	r.shutdown()
 
 	// Verify all connections are closed before closing DB
-	if conns := r.DB.Stats().OpenConnections; conns != 0 {
-		panic(fmt.Sprintf("found %d open sqlite connections", conns))
+	if conns := r.DB.Stats().Idle; conns != 0 {
```

Comment:
Why do we check for idle instead of open connections?

Reply:
Hmm maybe doing `if conns := r.DB.Stats().OpenConnections; conns != 1` would make more sense; Because I'll always have 1 connection open before I close it in `L102` with `r.DB.Close()`

Reply:
with `db.SetMaxIdleConns(0)` our DB will close all unused connections. If we still have opened connections it may mean that some query is still keeping it open/active and we missed `Close()` on statement or transaction.

When I faced this issue (with having 1 opened connection) it was because of that issue with migrator. After I used two DB instances I got 0 open connections...

Reply:
Ah I see, so we need to call close on all transactions or queries, I'll fix this, thanks for explaining :)

Reply:
Btw, Adam was able to find and fix leaking connections thanks to this open connections check. Initially I insisted we have to remove it %-)

Reply:
Yep. Having a non-zero amount of connections is a bug. Since this has come up twice now we should document the code to mention this (and perhaps the error message as well). 

See https://github.com/moov-io/paygate/pull/344 for more information

Reply:
I'll update this PR and include some comments documenting the behavior

Reply:
Follow-up, take a look at the changes in this commit: https://github.com/moov-io/base/pull/115/commits/c68e38a9ccfd858b717a684488bda417faf66176